src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -3881,12 +3881,7 @@ static NullableFlowState getResultState(BoundCollectionExpression node, Collecti
                 var collectionKind = ConversionsBase.GetCollectionExpressionTypeKind(this.compilation, collectionType, out var targetElementType);
                 if (collectionKind is CollectionExpressionTypeKind.CollectionBuilder)
                 {
-                    var createMethod = node.CollectionBuilderMethod;
-                    if (createMethod is not null)
-                    {
-                        var foundIterationType = _binder.TryGetCollectionIterationType((ExpressionSyntax)node.Syntax, collectionType, out targetElementType);
-                        Debug.Assert(foundIterationType);
-                    }
+                    _binder.TryGetCollectionIterationType((ExpressionSyntax)node.Syntax, collectionType, out targetElementType);
```

Comment:
Not related to this PR: I'm surprised that we only call `TryGetCollectionIterationType` for `CollectionBuilder` case. Shouldn't we also call it for `ImplementsIEnumerable` (since in that case we also won't have pre-determined the element type)?